fail to run kubectl set selector

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):BUG REPORT


**Kubernetes version** (use `kubectl version`):Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.2610+281053d66674e8-dirty", GitCommit:"281053d66674e870c93e4627a95689a4018f5f7b", GitTreeState:"dirty", BuildDate:"2016-12-29T06:28:44Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.2610+281053d66674e8-dirty", GitCommit:"281053d66674e870c93e4627a95689a4018f5f7b", GitTreeState:"dirty", BuildDate:"2016-12-29T06:28:44Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}


**What happened**:
```
# kubectl set selector
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x145f940, 0xc42000e0c0)
	/home/ubuntu/go/src/runtime/panic.go:500 +0x1a1
k8s.io/kubernetes/pkg/kubectl/cmd/set.getResourcesAndSelector(0x1ef82d0, 0x0, 0x0, 0xf, 0x0, 0x0, 0xbc6e97, 0xc420097b80, 0xc420432c30)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_selector.go:234 +0xd4
k8s.io/kubernetes/pkg/kubectl/cmd/set.(*SelectorOptions).Complete(0xc4201e2e60, 0x1e8cc40, 0xc42032a840, 0xc42041ed80, 0x1ef82d0, 0x0, 0x0, 0x1e71220, 0xc42002c010, 0x0, ...)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_selector.go:137 +0x827
k8s.io/kubernetes/pkg/kubectl/cmd/set.NewCmdSelector.func1(0xc42041ed80, 0x1ef82d0, 0x0, 0x0)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_selector.go:83 +0x85
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc42041ed80, 0x1ef82d0, 0x0, 0x0, 0xc42041ed80, 0x1ef82d0)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x439
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4204a3b00, 0xc4204a7700, 0x1, 0x1)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x367
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4204a3b00, 0xc42032a840, 0x1e711e0)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:38 +0xd5
main.main()
	/home/ubuntu/mygo/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:26 +0x22

```

--- a/pkg/kubectl/cmd/set/BUILD
+++ b/pkg/kubectl/cmd/set/BUILD
@@ -30,6 +30,7 @@ go_library(
         "//pkg/runtime:go_default_library",
         "//pkg/util/errors:go_default_library",
         "//pkg/util/strategicpatch:go_default_library",
+        "//pkg/util/validation:go_default_library",
         "//vendor:github.com/spf13/cobra",
     ],
 )

--- a/pkg/kubectl/cmd/set/set_selector.go
+++ b/pkg/kubectl/cmd/set/set_selector.go
@@ -29,6 +29,7 @@ import (
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/resource"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation"
 )
 
 // SelectorOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
@@ -77,7 +78,7 @@ func NewCmdSelector(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:     "selector (-f FILENAME | TYPE NAME) EXPRESSIONS [--resource-version=version]",
 		Short:   "Set the selector on a resource",
-		Long:    fmt.Sprintf(selectorLong),
+		Long:    fmt.Sprintf(selectorLong, validation.LabelValueMaxLength),
 		Example: selectorExample,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(options.Complete(f, cmd, args, out))
@@ -135,6 +136,9 @@ func (o *SelectorOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 func (o *SelectorOptions) Validate() error {
 	if len(o.resources) < 1 && cmdutil.IsFilenameEmpty(o.fileOptions.Filenames) {
 		return fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	}
+	if o.selector == nil {
+		return fmt.Errorf("one selector is required")
 	}
 	return nil
 }
@@ -213,9 +217,10 @@ func updateSelectorForObject(obj runtime.Object, selector metav1.LabelSelector) 
 
 // getResourcesAndSelector retrieves resources and the selector expression from the given args (assuming selectors the last arg)
 func getResourcesAndSelector(args []string) (resources []string, selector *metav1.LabelSelector, err error) {
-	if len(args) > 1 {
-		resources = args[:len(args)-1]
+	if len(args) == 0 {
+		return []string{}, nil, nil
 	}
+	resources = args[:len(args)-1]
 	selector, err = metav1.ParseToLabelSelector(args[len(args)-1])
 	return resources, selector, err
 }

--- a/pkg/kubectl/cmd/set/set_selector_test.go
+++ b/pkg/kubectl/cmd/set/set_selector_test.go
@@ -283,6 +283,13 @@ func TestGetResourcesAndSelector(t *testing.T) {
 				MatchExpressions: []metav1.LabelSelectorRequirement{},
 			},
 		},
+		{
+			name:          "no resource and selector",
+			args:          args{args: []string{}},
+			wantResources: []string{},
+			wantErr:       false,
+			wantSelector:  nil,
+		},
 	}
 	for _, tt := range tests {
 		gotResources, gotSelector, err := getResourcesAndSelector(tt.args.args)

